requests/signed_message_test.go

```diff
@@ -167,3 +167,35 @@ func TestSignedRequest_WithPemFile_NotRSAKeyShouldError(t *testing.T) {
 -----END RSA PUBLIC KEY-----`))
 	assert.ErrorContains(t, msg.Error, "not an RSA Private Key")
 }
+
+// TestAuthHeader tests the AuthHeader function.
```

Comment:
The AuthHeader function in request/signed_message.go is not used so let's remove the function there, and move this test so it tests the same function we have here https://github.com/getyoti/yoti-go-sdk/blob/SDK-2422-go-fix-code-smells/digitalidentity/requests/signed_message.go#L45